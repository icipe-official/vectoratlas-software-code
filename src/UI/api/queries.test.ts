import {
  referenceQuery,
  occurrenceQuery,
  speciesInformationById,
  upsertSpeciesInformationMutation,
  newsById,
  upsertNewsMutation,
  getAllNews,
} from './queries';

describe('occurrenceQuery', () => {
  let filters;

  beforeEach(() => {
    filters = {
      country: { value: null },
      species: { value: null },
      isLarval: { value: null },
      isAdult: { value: null },
      control: { value: null },
      season: { value: null },
      timeRange: {
        value: {
          start: null,
          end: null,
        },
      },
    };
  });

  it('returns the correct query without any filters', () => {
    expect(occurrenceQuery(0, 100, filters)).toMatchSnapshot();
  });

  it('includes countries if in filters', () => {
    filters.country.value = ['Kenya', 'Uganda'];
    expect(occurrenceQuery(0, 100, filters)).toMatchSnapshot();
  });

  it('converts time range if in filters', () => {
    filters.timeRange.value = {
      start: 1001,
      end: 2002,
    };
    expect(occurrenceQuery(0, 100, filters)).toMatchSnapshot();
  });
});

describe('referenceQuery', () => {
  it('returns the correct query', () => {
    expect(
      referenceQuery(10, 50, 'num_id', 'asc', 10, 100, 'filter')
    ).toMatchSnapshot();
  });
});

describe('species information', () => {
  it('speciesInformationById queries with the right id', () => {
    expect(speciesInformationById('123-ABC')).toContain(
      'speciesInformationById(id: "123-ABC")'
    );
    expect(speciesInformationById('123-ABC')).toMatchSnapshot();
  });

  it('upsertSpeciesInformationMutation builds the correct mutation if creating', () => {
    const speciesInformation = {
      name: 'test species',
      shortDescription: 'short description test',
      description: '# full description\n\nsome content',
      speciesImage: 'base64-image-ABCDEF',
    };

    const query = upsertSpeciesInformationMutation(speciesInformation);
    expect(query).not.toContain('id:');
    expect(query).toContain(`createEditSpeciesInformation(input: {
         
         name: "test species"
         shortDescription: "short description test"
         description: """# full description

some content"""
         speciesImage: "base64-image-ABCDEF"
      })`);
    expect(query).toMatchSnapshot();
  });

  it('upsertSpeciesInformationMutation builds the correct mutation if editing', () => {
    const speciesInformation = {
      id: '12345678',
      name: 'test species',
      shortDescription: 'short description test',
      description: '# full description\n\nsome content',
      speciesImage: 'base64-image-ABCDEF',
    };

    const query = upsertSpeciesInformationMutation(speciesInformation);
    expect(query).toContain(`createEditSpeciesInformation(input: {
         id: "12345678"
         name: "test species"
         shortDescription: "short description test"
         description: """# full description

some content"""
         speciesImage: "base64-image-ABCDEF"
      })`);
    expect(query).toMatchSnapshot();
  });
});

describe('news', () => {
  it('newsById queries with the right id', () => {
    expect(newsById('123-ABC')).toContain('newsById(id: "123-ABC")');
    expect(newsById('123-ABC')).toMatchSnapshot();
  });

  it('getAllNews does not ask for the full article', () => {
    expect(getAllNews()).not.toContain('article');
    expect(getAllNews()).toMatchSnapshot();
  });

  it('upsertNewsMutation builds the correct mutation if creating', () => {
    const news = {
      title: 'test news',
      summary: 'summary test',
      article: '# full article\n\nsome content',
      image: 'base64-image-ABCDEF',
    };

    const query = upsertNewsMutation(news);
    expect(query).not.toContain('id:');
    expect(query).toContain(`createEditNews(input: {
          
          title: "test news"
          summary: "summary test"
          article: """# full article

some content"""
          image: "base64-image-ABCDEF"
       })`);
    expect(query).toMatchSnapshot();
  });

  it('upsertNewsMutation builds the correct mutation if editing', () => {
    const news = {
      id: '12345678',
      title: 'test news',
      summary: 'summary test',
      article: '# full article\n\nsome content',
      image: 'base64-image-ABCDEF',
    };

    const query = upsertNewsMutation(news);
    expect(query).toContain(`createEditNews(input: {
          id: "12345678"
          title: "test news"
          summary: "summary test"
          article: """# full article

some content"""
          image: "base64-image-ABCDEF"
       })`);
    expect(query).toMatchSnapshot();
  });
});
