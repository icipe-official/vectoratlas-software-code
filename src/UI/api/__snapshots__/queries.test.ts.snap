// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`occurrenceQuery converts time range if in filters 1`] = `
"
query Occurrence {
   OccurrenceData(skip:0, take:100, filters: {startTimestamp:1001,endTimestamp:2002}, bounds: {locationWindowActive: false})
   {
      items {
         year_start
         id
         site {
            location
         }
         sample {
            n_all
         }
         recorded_species {
            species
         }
      }
      total
      hasMore
   }
}"
`;

exports[`occurrenceQuery includes countries if in filters 1`] = `
"
query Occurrence {
   OccurrenceData(skip:0, take:100, filters: {country:[\\"Kenya\\",\\"Uganda\\"]}, bounds: {locationWindowActive: false})
   {
      items {
         year_start
         id
         site {
            location
         }
         sample {
            n_all
         }
         recorded_species {
            species
         }
      }
      total
      hasMore
   }
}"
`;

exports[`occurrenceQuery returns the correct query without any filters 1`] = `
"
query Occurrence {
   OccurrenceData(skip:0, take:100, filters: {}, bounds: {locationWindowActive: false})
   {
      items {
         year_start
         id
         site {
            location
         }
         sample {
            n_all
         }
         recorded_species {
            species
         }
      }
      total
      hasMore
   }
}"
`;

exports[`referenceQuery returns the correct query 1`] = `
"
    query Reference{
        allReferenceData(skip:10, take:50, orderBy:\\"num_id\\", order:\\"asc\\", startId: 10, endId: 100, textFilter: \\"filter\\") {
         items{author
            article_title
            journal_title
            citation
            year
            published
            report_type
            v_data
            num_id

        }
    total
    hasMore
  }

    }
"
`;

exports[`species information speciesInformationById queries with the right id 1`] = `
"
   query {
      speciesInformationById(id: \\"123-ABC\\") {
        id
        name
        shortDescription
        description
      }
    }
    "
`;

exports[`species information upsertSpeciesInformationMutation builds the correct mutation if creating 1`] = `
"
   mutation {
      createEditSpeciesInformation(input: {
         
         name: \\"test species\\"
         shortDescription: \\"short description test\\"
         description: \\"\\"\\"# full description

some content\\"\\"\\"
         speciesImage: \\"ABC123\\"
      }) {
         name
         id
         description
         shortDescription
         speciesImage
      }
   }"
`;

exports[`species information upsertSpeciesInformationMutation builds the correct mutation if editing 1`] = `
"
   mutation {
      createEditSpeciesInformation(input: {
         id: \\"12345678\\"
         name: \\"test species\\"
         shortDescription: \\"short description test\\"
         description: \\"\\"\\"# full description

some content\\"\\"\\"
         speciesImage: \\"ABC123\\"
      }) {
         name
         id
         description
         shortDescription
         speciesImage
      }
   }"
`;
